overlord/snapstate/backend/link.go

```diff
@@ -67,8 +68,17 @@ func (b Backend) LinkSnap(info *snap.Info) error {
 	}
 
 	// XXX/TODO: this needs to be a task with proper undo and tests!
-	if err := boot.SetNextBoot(info); err != nil {
-		return err
+	if model != nil {
```

Comment:
given we moved some of the checking here, this should probably be guarded by !classic as well